doc/syntax.md

```diff
@@ -53,45 +334,84 @@ end
-: blabla
+  blocks are contained in this block. The block is closed with the `end` keyword.
+
+`parameters:`
+:  This block is composed of a list with variable declarations. This block is
+supposed to contain variables
+which remain constant during the concrete simulation, but they can vary among
+different simulations or instantiations neuron of the same neuron. The block
+is closed with the `end` keyword.
+
+`state:`
+: This block is composed of a list with variable declarations. These variables are supposed to
+model the time dependent  state of the neuron. Only variables form this block can be refined with
+differential equations. The block terminates with the `end` keyword.
+
+`internals:`
+:  This block is composed of a list with variable declarations. These variables
+are supposed to be constant during the simulation run. Therefore, its initialization
+expression can only reference parameter- or another internal-variables. The block
+is closed with the `end` keyword.
+
+`equations:`
+: This block is composed of shape definitions and differential equations. In will
+be explained in further details later in the manual. The block is closed with
+the `end` keyword.
+
+`input:`
+: The block is composed of a list of input ports. In will
+be explained in further details later in the manual. The block terminates with the `end` keyword.
+
+`output:`
+: Defines which kinds of events the neuron can fire. Currently, only `spike` is
+supported. No `end` is necessary at the end of the block.
+```
+output: spike
+```
+`update:`
+: Inside the block arbitrary code in the introduced programming language can be
+implemented. The block terminates with the `end` keyword.
+
+`function <name> ([<list_of_arguments>]) <return_type>?:`
+: `function ([<args>]`.
+The function has an unique name and a list of arguments between round brackets.
+Inside the block arbitrary code in the introduced
+programming language can be implemented. The function definitions
+terminates with the `end` keyword.
+```
+function <name>(<list_of_arguments>) <return_type>?:
+  <statements>
+end
 
+function devide(a real, b real) real:
+  return a/b
+end
+```
 
 ## Neuronal interactions
 
 ### Input
-    `inhibitory`
-    `excitatory`
-
+Input block is composed of input lines. Every input line of the form:
+```
+port_name <- inhibitory? excitatory? (spike | current)
+```
+There are three basic constilations for a valid input block.
+* It is composed of the single port, which is either `spike` or `current`,
+that receives as well positive weighted spikes as well as negative weighted spikes.
+* It is composed of two ports. The first port is an `inhibitory` port, the second
+port is an `excitatory` port. In that case, spikes are routed based on their
+weight. Positive weighted spikes go to the `excitatory` port, negative weighted
+spikes go to the `inhibitory` port. In that case both ports are mandatory.
+* There are more the one `spike` or `current` port. In that case, a multisynapse
```

Comment:
spikes are routed -> spikes IS routed?
to the `excitatory` port -> to THAN excitatory` port?
It sounds wrong for me.